static/templates/recipient_row.hbs

```diff
@@ -37,23 +37,26 @@
                 </a>
                 {{/each}}
 
-                {{! edit topic pencil icon }}
-                {{#if always_visible_topic_edit}}
-                    <i class="fa fa-pencil always_visible_topic_edit recipient_bar_icon" {{#unless realm_allow_message_editing}}style="display: none"{{/unless}} title="{{t 'Edit topic'}}" role="button" tabindex="0" aria-label="{{t 'Edit topic' }}"></i>
+                {{#if is_web_public_visitor }}
                 {{else}}
-                    {{#if on_hover_topic_edit}}
-                    <i class="fa fa-pencil on_hover_topic_edit recipient_bar_icon" {{#unless realm_allow_message_editing}}style="display: none"{{/unless}} title="{{t 'Edit topic'}}" role="button" tabindex="0" aria-label="{{t 'Edit topic' }}"></i>
+                    {{! edit topic pencil icon }}
+                    {{#if always_visible_topic_edit}}
+                        <i class="fa fa-pencil always_visible_topic_edit recipient_bar_icon" {{#unless realm_allow_message_editing}}style="display: none"{{/unless}} title="{{t 'Edit topic'}}" role="button" tabindex="0" aria-label="{{t 'Edit topic' }}"></i>
+                    {{else}}
+                        {{#if on_hover_topic_edit}}
+                        <i class="fa fa-pencil on_hover_topic_edit recipient_bar_icon" {{#unless realm_allow_message_editing}}style="display: none"{{/unless}} title="{{t 'Edit topic'}}" role="button" tabindex="0" aria-label="{{t 'Edit topic' }}"></i>
+                        {{/if}}
```

Comment:
I guess this is a case study of something where we could get away with a lot simpler change if we just made these elements `display: none` for web-public users.